### ISSUE TITLE
Creating a custom CacheManager bean loses the EntityManagerFactory bean's dependency on the cache manager

### ISSUE THREAD

**Issue by wilkinsona** [spring-projects/spring-boot#18107]
When a custom `CacheManager` is defined, all of `CacheAutoConfiguration` will back off, including `CacheManagerJpaDependencyConfiguration` which configures an entity manager factory beans to depend on the `cacheManager` bean. This is not consistent with other auto-configuration that configures dependencies between beans that do not back off when the target of the dependency is customized.

**Comment by wilkinsona** [comment]
There's a similar situation in `HazelcastJpaDependencyAutoConfiguration` as it backs off in the absence of a bean named `hazelcastInstance`. This one appears to be deliberate as there is a test that verifies the behaviour.

**Comment by snicoll** [comment]
> This is not consistent with other auto-configuration that configures dependencies between beans that do not back off when the target of the dependency is customized.

This one is a bit particular and certainly not ideal. There are cases where an application should bootstrap the cache before JPA kicks in (typical use case is to use the cache has a provider for Hibernate L2). Other users have reported the need to be able to bootstrap JPA before the cache kicks in (typically to use a `Repository` in their cache configuration if they want to use a loading cache.

In other words, the link between the two is certainly not as straightforward as the others. Doing it only for the cache manager we auto-configure is a way to get the 80% use case working out-of-the-box with a way for others to opt-out by creating their own `CacheManager`.
